great_expectations/datasource/fluent/sources.py

```diff
@@ -333,31 +334,101 @@ def pandas_default(self) -> PandasDatasource:
     def factories(self) -> List[str]:
         return list(self.__crud_registry.keys())
 
-    def _validate_datasource_type(
+    def _validate_current_datasource_type(
         self, name: str, datasource_type: Type[Datasource], raise_if_none: bool = True
     ) -> None:
         try:
-            old_datasource = self._data_context.get_datasource(name)
+            current_datasource = self._data_context.get_datasource(name)
         except ValueError as e:
             if raise_if_none:
                 raise ValueError(
                     f"There is no datasource {name} in the data context."
                 ) from e
-            old_datasource = None
-        if old_datasource and not isinstance(old_datasource, datasource_type):
+            current_datasource = None
+        if current_datasource and not isinstance(current_datasource, datasource_type):
             raise ValueError(
                 f"Trying to update datasource {name} but it is not the correct type. "
-                f"Expected {datasource_type} but got {type(old_datasource)}"
+                f"Expected {datasource_type} but got {type(current_datasource)}"
```

Comment:
```suggestion
                f"Expected {datasource_type.__name__} but got {type(current_datasource).__name__}"
```
`non-blocking`
Using the `class` `__name__` attribute might be slightly more readable.